lib/dex-oracle/driver.rb

```diff
@@ -210,7 +200,14 @@ def validate_output(full_cmd, full_output)
     logger.debug("Full output: #{full_output.inspect}")
     output_lines.reject! { |e| e.start_with?('WARNING:') }
     header = output_lines[0]
-    raise "app_process execution failure, output: '#{full_output}'" if header != OUTPUT_HEADER
+    if header != OUTPUT_HEADER
+      print "app_process execution failure, output: '#{full_output}'"
+      exception = adb("shell cat #{@driver_dir}/od-exception.txt").strip
+      unless exception.end_with?('No such file or directory')
+        adb("shell rm #{@driver_dir}/od-exception.txt")
+        raise exception
+      end
```

Comment:
This looks like it misses the condition where you have an unexpected header and there's no od-exception.txt, i.e. a catastrophic failure.